client/src/components/employee/AddEmployeeDialogue.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div>
+      <div class="overlay" v-if="value"></div>
+
+      <div class="modal" v-if="value">
+        <h3>Add Employee</h3>
+
+        <form @submit.prevent="addEmployee">
+          <md-field :class="idMessageClass">
+            <label for="employee_id">ID</label>
+            <md-input id="employee_id" required v-model="id" maxlength="6"></md-input>
+            <span class="md-error">{{this.idErrorMessage}}</span>
+          </md-field>
+
+          <md-field :class="nameMessageClass">
+            <label for="employee_name">Full ame</label>
+            <md-input id="employee_name" required v-model="name" maxlength="50"></md-input>
+            <span class="md-error">{{this.nameErrorMessage}}</span>
+          </md-field>
+
+          <md-button class="md-raised" type="cancel" @click="$emit('input', false)">CANCEL</md-button>
+          <md-button class="md-raised md-primary" :disabled="!formValidated" type="submit">SUBMIT</md-button>
+        </form>
+      </div>
+
+      <md-snackbar md-position="left" :md-duration="snackbarDuration" :md-active.sync="showSnackbar" md-persistent>
+        <span>{{snackbarMessage}}</span>
+      </md-snackbar>
+  </div>
+</template>
+
+<script>
+import { employeeService } from '../../service/employee'
+export default {
+  props: ['value'],
```

Comment:
This is very odd... I don't understand why you haven't used a `v-model` for the entire form?

Reply:
I used value prop and input emit in this component and v-model is used by it's parent component.